sklearn/mixture/gaussian_mixture.py

```diff
@@ -0,0 +1,677 @@
+        `weights_` will not exist before a call to fit.
+
+    means_ : array, shape (`n_components`, `n_features`)
+        The mean of each mixture component.
+        `means_` will not exist before a call to fit.
+
+    covariances_ : array
+        The covariance of each mixture component.
+        The shape depends on `covariance_type`::
+            (n_components,)                        if 'spherical',
+            (n_features, n_features)               if 'tied',
+            (n_components, n_features)             if 'diag',
+            (n_components, n_features, n_features) if 'full'
+        `covariances_` will not exist before a call to fit.
+
+    converged_ : bool
+        True when convergence was reached in fit(), False otherwise.
+        `converged_` will not exist before a call to fit.
+
+    initialized_ : bool
+        True when fit() has been applied at least once, False otherwise.
+        `initialized_` will not exist before a call to fit.
+    """
+
+    def __init__(self, n_components=1, covariance_type='full',
+                 random_state=None, tol=1e-6, reg_covar=0.,
+                 max_iter=100, n_init=1, init_params='kmeans',
+                 weights_init=None, means_init=None, covariances_init=None,
+                 warm_start=False, verbose=0, verbose_interval=10):
+        super(GaussianMixture, self).__init__(
+            n_components=n_components, random_state=random_state, tol=tol,
+            reg_covar=reg_covar, max_iter=max_iter, n_init=n_init,
+            init_params=init_params, warm_start=warm_start,
+            verbose=verbose, verbose_interval=verbose_interval)
+
+        self.covariance_type = covariance_type
+        self.weights_init = weights_init
+        self.means_init = means_init
+        self.covariances_init = covariances_init
+
+    def _define_parameter_shape(self, n_features):
+        """Define the shape of the parameters."""
+        cov_shape = {'full': (self.n_components, n_features,
+                              n_features),
+                     'tied': (n_features, n_features),
+                     'diag': (self.n_components, n_features),
+                     'spherical': (self.n_components,)}
+
+        param_shape = {'weights': (self.n_components,),
+                       'means': (self.n_components, n_features),
+                       'covariances': cov_shape[self.covariance_type]}
+
+        return param_shape
+
+    def _check_parameters(self, X):
+        """Check the Gaussian mixture parameters are well defined."""
+        if self.covariance_type not in ['spherical', 'tied', 'diag', 'full']:
+            raise ValueError("Invalid value for 'covariance_type': %s "
+                             "'covariance_type' should be in "
+                             "['spherical', 'tied', 'diag', 'full']"
+                             % self.covariance_type)
+
+        param_shape = self._define_parameter_shape(int(X.shape[1]))
+
```

Comment:
To solve the Travis problem (with Python-2.7 -- x64), I have put `int(X.shape[1])`.
I don't like it but I haven't find a better solution.